src/lib/cloud-manager-hoc.jsx

```diff
@@ -124,23 +121,30 @@ const cloudManagerHOC = function (WrappedComponent) {
     }
 
     CloudManager.propTypes = {
-        canSave: PropTypes.bool.isRequired,
+        canModifyCloudData: PropTypes.bool.isRequired,
```

Comment:
note that I switched the boolean value of shouldNotModifyCloudData canModifyCloudData.

There's potential confusion with `hasCloudPermission`, which in www is set to `isScratcher`; I have ideas for further simplification here, but I think we should take it one step at a time.